### ISSUE TITLE
Quoted PutLink crashes

### ISSUE THREAD

**Issue by ngeiswei** [opencog/atomspace#2338]
The following
```scheme
(Quote (Put
    (Unquote (Variable "$vardecl"))
    (Unquote (Variable "$body"))
    (Unquote (Variable "$args"))))
```
produces
```
Segmentation fault (core dumped)
```

**Comment by ngeiswei** [comment]
BTW, the following
```scheme
(Quote (Put
    (Unquote (Variable "$body"))
    (Unquote (Variable "$args"))))
```
does NOT produce a segfault.

**Comment by ngeiswei** [comment]
I've added a (disabled) unit test to reproduce the problem

https://github.com/opencog/atomspace/blob/d8e3a92833e168c13633a54dbfdc22158185397f/tests/atoms/core/PutLinkUTest.cxxtest#L2244-L2259

**Comment by linas** [comment]
Hmm sometimes github auto-closes issues, and sometimes it doesn't.  If your pull req had said "this fixes #2338" then this would have auto-closed when it got merged.
